### PR TITLE
Implement split function that is an inverse function of stack

### LINKED ISSUE #1344: Inverse function of stack
I often concatenate and split `Variable`s with `stack` and `split_axis` function. As `split_axis` is not an inverse of `stack`, I need to call `reshape` for split variables.
I need an inverse function of `stack`, that split a `(N, S, T)`-shape variable to a list of N `(S, T)`-shape variables. I don't know its corresponding function in NumPy.

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented an inverse function of `stack`. We need to discuss about its name.
fix #1344

### CODE REVIEW

**Comment by delta2323** [comment]
I agree with you in that `split` is most intuitive and simple name. But, as you may know it, the behavior is different from `split` functioin in NumPy (rather, `F.split_axis` is similar to `numpy.split`). One possible solution is to use the term similar to `split` like `separate`.

**Comment by unnonouno** [comment]
Oh, `separate` looks good.

**Comment by unnonouno** [comment]
I wrote docstring.

File: chainer/functions/array/separate.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def separate(x, axis=0):
+    """Separates an array along a given axis.
+
+    This function separates an array along a given axis. For example, shape of
+    an array is ``(2, 3, 4)``. When it separates the array with ``axis=1``, it
+    returns three ``(2, 4)`` arrays.
+
+    This function is an invers of :func:`chainer.functions.stack`.
```
Comment (mattya) [inline chainer/functions/array/separate.py]: invers -> inverse

**Comment by mattya** [comment]
LGTM except for the typo!

**Comment by unnonouno** [comment]
I fixed it.

**Pull request merged**